packages/algob/src/lib/nft.ts

```diff
@@ -0,0 +1,106 @@
+import type { Account } from "algosdk";
+
+import { AlgobDeployer, NFT, SignType, SSCStateSchema } from "../types";
+import { readGlobalStateSSC, readLocalStateSSC } from "./status";
+import { executeTransaction } from "./tx";
+
+// generate random 32 char hash
+function _NFTToken (): string {
+  return Array(32).fill(0).map(x => Math.random().toString(36).charAt(2)).join('');
+}
+
+/**
+ * Description: This function creates a new Non-Fungible-Token
+ * @param deployer AlgobDeployer
+ * @param sender Account from which call needs to be made
+ * @param appId application ID
+ * @param nft contains name of nft
+ */
+export async function createNewNFT (
+  deployer: AlgobDeployer,
+  creater: Account,
+  appId: number,
+  nft: NFT
+): Promise<void> {
+  const globalState = await readGlobalStateSSC(deployer, creater.addr, appId) as SSCStateSchema[];
+  if (globalState === undefined) {
+    console.warn(`${creater.addr} is not the smart contract admin`);
+    return;
+  }
+
+  for (const g of globalState) {
+    const key = Buffer.from(g.key, 'base64').toString();
+    if (key === nft.name) {
+      throw new Error(`Non Fungible Token with name ${nft.name} already exists`);
+    }
+  }
+
+  nft.url = _NFTToken();
+
+  const appArgs = [];
+  appArgs.push(new Uint8Array(Buffer.from("create")));
+  appArgs.push(new Uint8Array(Buffer.from(nft.name)));
+  appArgs.push(new Uint8Array(Buffer.from(nft.url)));
```

Comment:
I think using simple Buffer will work here as well ( we don't need to convert to `Uint8Array`). Can you check that please?

Reply:
Simple buffer doesn't work here. The `application call` in `algosdk` expects appArgs to be an array of `Uint8Array`.
![image](https://user-images.githubusercontent.com/33264364/99881502-3c87e480-2c40-11eb-90e9-4fd366c8bdb0.png)